### PR TITLE
Add precommit hook and github action

### LINKED ISSUE #19: Add pre-commit hook
From https://github.com/drift-org/.github/issues/1:

Add precommit hook that runs `make clean`. Include setup instructions.

### PR DESCRIPTION
**Pull request by brandonLi8**
# Contributors

@brandonLi8 

# Relevant issue

Closes #19

# Summary of change

- Added pre-commit hook that runs `make clean`
- Added make command to install the hooks (`make setup-hooks`)
- Added github action that:
   + checks that `make clean` doesn't change any files (checks that formatting is correct)
   + runs unit tests (currently don't have any, so always succeeds, but unit tests will be added in #18)

# Testing/Verification

- Tested on my commit that the hook runs (it logs output)
- Note that the action won't run on this pr, since it hasn't been merged. I tested that the format check on the action is correct and properly errors when formatting changes were detected.

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - uses: actions/checkout@v2    # Checkout the source code of the project.
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with: |
+        go-version: 1.15
+
+    - name: formatting             # Ensure formatting is correct.
+      run: |
+        make clean
```
Comment (anjanbharadwaj) [inline .github/workflows/CI.yaml:33]: If we have a pre-commit hook that's already running make clean before any commit, then there's no need to include it again in the workflow.

File: .githooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+#!————————————————————————————————————————————————————————————————————————————*!
+# Git pre-commit hook.
+#
+# Run `make setup-hooks` to install.
+#!————————————————————————————————————————————————————————————————————————————*!
+
+echo "running pre-commit hook..."
+
+# Format files.
+make clean
```
Comment (anjanbharadwaj) [inline .githooks/pre-commit]: Need to re-add the files that have been formatted so that they're included in the commit - a simple git add . should do

**Review by anjanbharadwaj — CHANGES_REQUESTED** [review]
Some small changes here & there

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (brandonLi8) [reply .github/workflows/CI.yaml:33]: I think the workflow guarantees it. There are a couple of thing it guards against:
  1. Forgetting to set up hooks
  2. Bypassing hooks with `-n`

File: .githooks/pre-commit (hunk 2 above)
Comment (brandonLi8) [reply .githooks/pre-commit]: good catch

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (anjanbharadwaj) [reply .github/workflows/CI.yaml:33]: Bro no one's gonna bypass hooks. Anyways, doesn't hurt that much to have it (slows down the CI workflow but not by much) so its fine for now.

**Review by anjanbharadwaj — APPROVED** [review]

**Review by aidenszeto — APPROVED** [review]

**Pull request merged**